helper/schema/schema.go

```diff
@@ -1431,6 +1480,79 @@ func (m schemaMap) validateConflictingAttributes(
 	return nil
 }
 
+func removeDuplicates(elements []string) []string {
+	encountered := make(map[string]struct{}, 0)
+	result := []string{}
+
+	for v := range elements {
+		if _, ok := encountered[elements[v]]; !ok {
+			encountered[elements[v]] = struct{}{}
+			result = append(result, elements[v])
+		}
+	}
+
+	return result
+}
+
+func validateExactlyOneAttributes(
+	k string,
+	schema *Schema,
+	c *terraform.ResourceConfig) error {
+
+	if len(schema.ExactlyOneOf) == 0 {
+		return nil
+	}
+
+	allKeys := removeDuplicates(append(schema.ExactlyOneOf, k))
+	sort.Strings(allKeys)
```

Comment:
do we need to sort?

Reply:
This is for nicer error messages. These have the potential to stack up on one another so if we sort them, they will line up nicely with one another